1.17.1 Towny plugin does not work.

### What steps will reproduce the problem?

1. Download the plugin 2. Add the plugin to the plugins folder 3. Start the server

### What is the expected output?

Towny plugin was not enabled, I was waiting for it to be enabled.

### Towny version

0.100.2.7

### Server version

1.17.1

### Please use Pastebin.com to link the following files

1. Your full server startup from the logs\latest.log: 
```
[19:01:22] [main/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[19:01:23] [main/INFO]: Reloading ResourceManager: Default, bukkit
[19:01:25] [Worker-Main-1/INFO]: Loaded 7 recipes
[19:01:30] [Server thread/INFO]: Starting minecraft server version 1.17.1
[19:01:31] [Server thread/WARN]: ****************************
[19:01:31] [Server thread/WARN]: YOU ARE RUNNING THIS SERVER AS AN ADMINISTRATIVE OR ROOT USER. THIS IS NOT ADVISED.
[19:01:31] [Server thread/WARN]: YOU ARE OPENING YOURSELF UP TO POTENTIAL RISKS WHEN DOING THIS.
[19:01:31] [Server thread/WARN]: FOR MORE INFORMATION, SEE https://madelinemiller.dev/blog/root-minecraft-server/
[19:01:31] [Server thread/WARN]: ****************************
[19:01:31] [Server thread/INFO]: Loading properties
[19:01:31] [Server thread/INFO]: This server is running Paper version git-Paper-411 (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: 6625db3 on ver/1.17.1)
[19:01:31] [Server thread/INFO]: Using 4 threads for Netty based IO
[19:01:31] [Server thread/INFO]: Server Ping Player Sample Count: 12
[19:01:31] [Server thread/INFO]: Default game type: SURVIVAL
[19:01:31] [Server thread/INFO]: Generating keypair
[19:01:31] [Server thread/INFO]: Starting Minecraft server on *:25565
[19:01:32] [Server thread/INFO]: Using default channel type
[19:01:32] [Server thread/INFO]: Paper: Using Java 11 compression from Velocity.
[19:01:32] [Server thread/INFO]: Paper: Using Java cipher from Velocity.
[19:01:34] [Server thread/ERROR]: [STDERR] [org.bukkit.craftbukkit.v1_17_R1.legacy.CraftLegacy] Initializing Legacy Material Support. Unless you have legacy plugins and/or data this is a bug!
[19:01:50] [Server thread/WARN]: Legacy plugin WebSender v1.0.2 does not specify an api-version.
[19:01:51] [Server thread/INFO]: [GroupManager] Loading 6 libraries... please wait
[19:01:51] [Server thread/INFO]: [GroupManager] Loaded library C:\Users\Administrator\Desktop\Pagist Towny\libraries\org\postgresql\postgresql\42.4.2\postgresql-42.4.2.jar
[19:01:51] [Server thread/INFO]: [GroupManager] Loaded library C:\Users\Administrator\Desktop\Pagist Towny\libraries\org\checkerframework\checker-qual\3.5.0\checker-qual-3.5.0.jar
[19:01:51] [Server thread/INFO]: [GroupManager] Loaded library C:\Users\Administrator\Desktop\Pagist Towny\libraries\com\h2database\h2\2.1.214\h2-2.1.214.jar
[19:01:51] [Server thread/INFO]: [GroupManager] Loaded library C:\Users\Administrator\Desktop\Pagist Towny\libraries\mysql\mysql-connector-java\8.0.30\mysql-connector-java-8.0.30.jar
[19:01:51] [Server thread/INFO]: [GroupManager] Loaded library C:\Users\Administrator\Desktop\Pagist Towny\libraries\com\google\protobuf\protobuf-java\3.19.4\protobuf-java-3.19.4.jar
[19:01:51] [Server thread/INFO]: [GroupManager] Loaded library C:\Users\Administrator\Desktop\Pagist Towny\libraries\org\mariadb\jdbc\mariadb-java-client\3.0.7\mariadb-java-client-3.0.7.jar
[19:01:51] [Server thread/INFO]: [GroupManager] Loaded library C:\Users\Administrator\Desktop\Pagist Towny\libraries\com\zaxxer\HikariCP\5.0.1\HikariCP-5.0.1.jar
[19:01:51] [Server thread/INFO]: [GroupManager] Loaded library C:\Users\Administrator\Desktop\Pagist Towny\libraries\org\slf4j\slf4j-api\2.0.0\slf4j-api-2.0.0.jar
[19:01:51] [Server thread/ERROR]: Could not load 'plugins\Dynmap-Towny-1.1.2.jar' in folder 'plugins'
org.bukkit.plugin.UnknownDependencyException: Unknown/missing dependency plugins: [dynmap]. Please download and install these plugins to run 'Dynmap-Towny'.
	at org.bukkit.plugin.SimplePluginManager.loadPlugins(SimplePluginManager.java:290) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.loadPlugins(CraftServer.java:419) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:287) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1220) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[patched_1.17.1.jar:git-Paper-411]
	at java.lang.Thread.run(Thread.java:842) ~[?:?]
[19:01:52] [Server thread/INFO]: [WebSender] Loading WebSender v1.0.2
[19:01:52] [Server thread/INFO]: [GroupManager] Loading GroupManager v3.2 (Phoenix)
[19:01:52] [Server thread/INFO]: [GroupManager] Dependencies: OK
[19:01:52] [Server thread/INFO]: [Vault] Loading Vault v1.7.3-b131
[19:01:52] [Server thread/INFO]: [GWarp] Loading GWarp v[2.13.0.2]
[19:01:52] [Server thread/INFO]: [Multiverse-Core] Loading Multiverse-Core v4.3.1-b861
[19:01:52] [Server thread/INFO]: [ProtocolLib] Loading ProtocolLib v5.2.0-SNAPSHOT-679
[19:01:55] [Server thread/INFO]: [LagAssist] Loading LagAssist v2.29.1
[19:01:55] [Server thread/INFO]: [SetHomes] Loading SetHomes v1.3.0
[19:01:55] [Server thread/INFO]: [WorldEdit] Loading WorldEdit v7.2.20+6667-f0f1e66
[19:01:57] [Server thread/INFO]: Got request to register class com.sk89q.worldedit.bukkit.BukkitServerInterface with WorldEdit [com.sk89q.worldedit.extension.platform.PlatformManager@1c38fee8]
[19:01:57] [Server thread/INFO]: [AuthMe] Loading AuthMe v5.6.0-SNAPSHOT-b2623
[19:01:57] [Server thread/INFO]: [Towny] Loading Towny v0.100.2.6
[19:01:57] [Server thread/INFO]: [WorldGuard] Loading WorldGuard v7.0.7+216b061
[19:01:57] [Server thread/INFO]: [Citizens] Loading Citizens v2.0.28-SNAPSHOT (build 2346)
[19:01:57] [Server thread/INFO]: [LockettePro] Loading LockettePro v2.11.1
[19:01:57] [Server thread/INFO]: [TownyCultures] Loading TownyCultures v2.0.1
[19:01:57] [Server thread/INFO]: [BetterRTP] Loading BetterRTP v3.6.12
[19:01:57] [Server thread/INFO]: [QuickShop] Loading QuickShop v5.1.2.5
[19:01:57] [Server thread/INFO]: [QuickShop] QuickShop Reremake - Early boot step - Booting up
[19:01:58] [Server thread/INFO]: [QuickShop] [OK] Signature Verify
[19:01:58] [Server thread/INFO]: [QuickShop] [OK] Potential Infection Characteristics Check
[19:01:58] [Server thread/INFO]: [QuickShop] [OK] Plugin Manifest Check
[19:01:58] [Server thread/INFO]: [QuickShop] Reading the configuration...
[19:01:59] [Server thread/INFO]: [QuickShop] Loading messages translation over-the-air (this may need take a while).
[19:01:59] [Server thread/INFO]: [QuickShop] Translation over-the-air platform selected: Crowdin
[19:01:59] [Server thread/INFO]: [QuickShop] Checking for translation updates, this may need a while...
[19:02:04] [Server thread/INFO]: [QuickShop] Loading up integration modules.
[19:02:04] [Server thread/INFO]: [QuickShop] QuickShop Reremake - Early boot step - Complete
[19:02:04] [Server thread/INFO]: [SiegeWar] Loading SiegeWar v2.10.0
[19:02:04] [Server thread/INFO]: [TownyFlight] Loading TownyFlight v1.10.2
[19:02:04] [Server thread/INFO]: Server permissions file permissions.yml is empty, ignoring it
[19:02:04] [Server thread/INFO]: [Vault] Enabling Vault v1.7.3-b131
[19:02:04] [Server thread/WARN]: [Vault] Loaded class org.anjocaido.groupmanager.GroupManager from GroupManager v3.2 (Phoenix) which is not a depend, softdepend or loadbefore of this plugin.
[19:02:04] [Server thread/INFO]: [Vault] [Permission] GroupManager found: Waiting
[19:02:04] [Server thread/INFO]: [Vault] [Permission] SuperPermissions loaded as backup permission system.
[19:02:04] [Server thread/INFO]: [Vault] [Chat] GroupManager found: Waiting
[19:02:05] [Server thread/INFO]: [Vault] Enabled Version 1.7.3-b131
[19:02:05] [Server thread/INFO]: [ProtocolLib] Enabling ProtocolLib v5.2.0-SNAPSHOT-679
[19:02:05] [Server thread/INFO]: [WorldEdit] Enabling WorldEdit v7.2.20+6667-f0f1e66
[19:02:05] [Server thread/INFO]: Registering commands with com.sk89q.worldedit.bukkit.BukkitServerInterface
[19:02:05] [Server thread/WARN]: [WorldEdit] Loaded class org.anjocaido.groupmanager.dataholder.worlds.WorldsHolder from GroupManager v3.2 (Phoenix) which is not a depend, softdepend or loadbefore of this plugin.
[19:02:05] [Server thread/INFO]: WEPIF: Vault detected! Using Vault for permissions
[19:02:05] [Server thread/INFO]: Using com.sk89q.worldedit.bukkit.adapter.impl.v1_17_R1_2.PaperweightAdapter as the Bukkit adapter
[19:02:07] [Server thread/WARN]: **** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!
[19:02:07] [Server thread/WARN]: The server will make no attempt to authenticate usernames. Beware.
[19:02:07] [Server thread/WARN]: While this makes the game possible to play without internet access, it also opens up the ability for hackers to connect with any username they choose.
[19:02:07] [Server thread/WARN]: To change this, set "online-mode" to "true" in the server.properties file.
[19:02:07] [Server thread/INFO]: Preparing level "world"
[19:02:09] [Server thread/INFO]: Preparing start region for dimension minecraft:overworld
[19:02:10] [Server thread/INFO]: Time elapsed: 434 ms
[19:02:10] [Server thread/INFO]: Preparing start region for dimension minecraft:the_nether
[19:02:10] [Server thread/INFO]: Time elapsed: 221 ms
[19:02:10] [Server thread/INFO]: Preparing start region for dimension minecraft:the_end
[19:02:10] [Server thread/INFO]: Time elapsed: 185 ms
[19:02:10] [Server thread/INFO]: [WebSender] Enabling WebSender v1.0.2*
[19:02:10] [Server thread/INFO]: [WebSender] PLUGIN DISABLED! PLEASE CHANGE PASSWORD FROM CONFIG!
[19:02:10] [Server thread/INFO]: [GroupManager] Enabling GroupManager v3.2 (Phoenix)
[19:02:10] [Server thread/INFO]: [GroupManager] OfflinePlayers cached ( 1 ).
[19:02:10] [Server thread/INFO]: [GroupManager] World Found: world
[19:02:10] [Server thread/INFO]: [GroupManager] Superperms support enabled.
[19:02:10] [Server thread/INFO]: [GroupManager] Scheduled Data Saving is set for every 10 minutes!
[19:02:10] [Server thread/INFO]: [GroupManager] Backups will be retained for 24 hours!
[19:02:10] [Server thread/INFO]: [GroupManager] DataSource - YAML
[19:02:10] [Server thread/INFO]: [GroupManager] version 3.2 (Phoenix) is enabled!
[19:02:10] [Server thread/INFO]: WEPIF: Vault detected! Using Vault for permissions
[19:02:10] [Server thread/INFO]: [Vault][Permission] GroupManager hooked.
[19:02:10] [Server thread/INFO]: [Vault] [Vault][Chat] GroupManager - Chat hooked.
[19:02:10] [Server thread/INFO]: [GWarp] Enabling GWarp v[2.13.0.2]
[19:02:10] [Server thread/INFO]: [GWarp] The Plugin was successfully enabled.
[19:02:11] [Server thread/INFO]: [Multiverse-Core] Enabling Multiverse-Core v4.3.1-b861
[19:02:11] [Server thread/WARN]: [Multiverse-Core] "Multiverse-Core v4.3.1-b861" has registered a listener for org.bukkit.event.entity.EntityCreatePortalEvent on method "public void com.onarandombox.MultiverseCore.listeners.MVPortalListener.entityPortalCreate(org.bukkit.event.entity.EntityCreatePortalEvent)", but the event is Deprecated. "Server performance will be affected"; please notify the authors [dumptruckman, Rigby, fernferret, lithium3141, main--].
[19:02:11] [Server thread/INFO]: [Multiverse-Core] We are aware of the warning about the deprecated event. There is no alternative that allows us to do what we need to do and performance impact is negligible. It is safe to ignore.
[19:02:11] [Server thread/INFO]: [GroupManager] New world detected...
[19:02:11] [Server thread/INFO]: [GroupManager] Creating data for: Lobby
[19:02:11] [Server thread/INFO]: [GroupManager] Don't forget to configure/mirror this world in config.yml.
[19:02:11] [Server thread/INFO]: Preparing start region for dimension minecraft:lobby
[19:02:11] [Server thread/INFO]: Time elapsed: 170 ms
[19:02:11] [Server thread/INFO]: [Multiverse-Core] 4 - World(s) loaded.
[19:02:11] [Server thread/WARN]: [Multiverse-Core] Buscript failed to load! The script command will be disabled! If you would like not to see this message, use `/mv conf enablebuscript false` to disable Buscript from loading.
[19:02:11] [Server thread/INFO]: [Multiverse-Core] Version 4.3.1-b861 (API v24) Enabled - By dumptruckman, Rigby, fernferret, lithium3141 and main--
[19:02:11] [Server thread/INFO]: [LagAssist] Enabling LagAssist v2.29.1
[19:02:11] [Server thread/INFO]: LagAssist » Enabling Systems:
[19:02:11] [Server thread/INFO]:     [?] Economy manager
[19:02:12] [Server thread/INFO]:     [?] Smart Mob Tools.
[19:02:12] [Server thread/INFO]:     [?] MicroFeatures.
[19:02:12] [Server thread/INFO]:     [?] Hopper Manager.
[19:02:12] [Server thread/INFO]:     [?] Redstone Culler.
[19:02:12] [Server thread/INFO]:     [?] Physics-Tweaker.
[19:02:12] [Server thread/INFO]:     [?] Lag Monitor.
[19:02:12] [Server thread/INFO]:     [?] MapVisualizer.
[19:02:12] [Server thread/INFO]:     [?] Chunk Analyser.
[19:02:12] [Server thread/INFO]:     [?] Chunk Limiter.
[19:02:12] [Server thread/INFO]:     [?] StatsAnalyse.
[19:02:12] [Server thread/INFO]:     [?] Spawner Manager.
[19:02:12] [Server thread/INFO]:     [?] GUI.
[19:02:12] [Server thread/INFO]:     [?] Injecting PacketListener.
[19:02:12] [Server thread/INFO]:     [?] API Tools.
[19:02:12] [Server thread/INFO]: [SetHomes] Enabling SetHomes v1.3.0
[19:02:12] [Server thread/INFO]: [SetHomes] Luck perms was not found! Reverting to vault...
[19:02:12] [Server thread/INFO]: [SetHomes] Hooked into Vault!
[19:02:12] [Server thread/INFO]: [AuthMe] Enabling AuthMe v5.6.0-SNAPSHOT-b2623
[19:02:12] [Server thread/INFO]: [AuthMe] Connection arguments loaded, Hikari ConnectionPool ready!
[19:02:12] [Server thread/ERROR]: [AuthMe] [STDERR] [Server thread] INFO fr.xephi.authme.libs.com.zaxxer.hikari.HikariDataSource - AuthMeMYSQLPool - Starting...
[19:02:12] [Server thread/WARN]: Nag author(s): '[sgdc3, games647, Hex3l, krusic22]' of 'AuthMe' about their usage of System.out/err.print. Please use your plugin's logger instead (JavaPlugin#getLogger).
[19:02:13] [Server thread/ERROR]: [AuthMe] [STDERR] [Server thread] INFO fr.xephi.authme.libs.com.zaxxer.hikari.HikariDataSource - AuthMeMYSQLPool - Start completed.
[19:02:14] [Server thread/INFO]: [AuthMe] MySQL setup finished
[19:02:14] [Server thread/INFO]: [AuthMe] Hooked into Vault!
[19:02:14] [Server thread/INFO]: [AuthMe] Hooked successfully into Multiverse-Core
[19:02:15] [Server thread/WARN]: [AuthMe] Failed to compile pattern '[a-zA-Z0-9_-.]*' - defaulting to allowing everything
[19:02:15] [Server thread/INFO]: [AuthMe] AuthMe 5.6.0-SNAPSHOT build n.2623 successfully enabled!
[19:02:15] [Server thread/INFO]: [Towny] Enabling Towny v0.100.2.6
[19:02:15] [Server thread/INFO]: ====================      Towny      ========================
[19:02:15] [Server thread/ERROR]: Error occurred while enabling Towny v0.100.2.6 (Is it up to date?)
java.lang.NoClassDefFoundError: org/apache/commons/io/FilenameUtils
	at com.palmergames.compress.utils.FileNameUtils.getBaseName(FileNameUtils.java:48) ~[?:?]
	at com.palmergames.bukkit.towny.object.TranslationLoader.lambda$getLangFileNamesFromPlugin$0(TranslationLoader.java:196) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178) ~[?:?]
	at java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?]
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?]
	at com.palmergames.bukkit.towny.object.TranslationLoader.getLangFileNamesFromPlugin(TranslationLoader.java:197) ~[?:?]
	at com.palmergames.bukkit.towny.object.TranslationLoader.loadTranslationsIntoMemory(TranslationLoader.java:164) ~[?:?]
	at com.palmergames.bukkit.towny.object.Translation.loadTranslationRegistry(Translation.java:47) ~[?:?]
	at com.palmergames.bukkit.towny.Towny.loadLocalization(Towny.java:276) ~[?:?]
	at com.palmergames.bukkit.towny.Towny.loadFoundation(Towny.java:227) ~[?:?]
	at com.palmergames.bukkit.towny.Towny.onEnable(Towny.java:137) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:561) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:475) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:733) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:317) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1220) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[patched_1.17.1.jar:git-Paper-411]
	at java.lang.Thread.run(Thread.java:842) ~[?:?]
Caused by: java.lang.ClassNotFoundException: org.apache.commons.io.FilenameUtils
	at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:151) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:103) ~[patched_1.17.1.jar:git-Paper-411]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	... 27 more
[19:02:15] [Server thread/INFO]: [Towny] Disabling Towny v0.100.2.6
[19:02:15] [Server thread/INFO]: ==============================================================
[19:02:15] [Server thread/INFO]: [Towny] Finishing Universe Tasks...
[19:02:15] [Server thread/INFO]: [Towny] Version: 0.100.2.6 - Plugin Disabled
[19:02:15] [Server thread/INFO]: =============================================================
[19:02:15] [Server thread/INFO]: [WorldGuard] Enabling WorldGuard v7.0.7+216b061
[19:02:16] [Server thread/WARN]: [WorldGuard] Loaded class fr.xephi.authme.libs.com.mysql.cj.jdbc.Driver from AuthMe v5.6.0-SNAPSHOT-b2623 which is not a depend, softdepend or loadbefore of this plugin.
[19:02:16] [Server thread/ERROR]: No TreeType mapping for TreeGenerator.TreeType.MANGROVE
[19:02:16] [Server thread/ERROR]: No TreeType mapping for TreeGenerator.TreeType.TALL_MANGROVE
[19:02:16] [Server thread/ERROR]: No TreeType mapping for TreeGenerator.TreeType.CHERRY
[19:02:16] [Server thread/INFO]: [WorldGuard] (world) TNT ignition is PERMITTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] (world) Lighters are PERMITTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] (world) Lava fire is PERMITTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] (world) Fire spread is UNRESTRICTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] Loaded configuration for world 'world'
[19:02:16] [Server thread/INFO]: [WorldGuard] (world_nether) TNT ignition is PERMITTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] (world_nether) Lighters are PERMITTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] (world_nether) Lava fire is PERMITTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] (world_nether) Fire spread is UNRESTRICTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] Loaded configuration for world 'world_nether'
[19:02:16] [Server thread/INFO]: [WorldGuard] (world_the_end) TNT ignition is PERMITTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] (world_the_end) Lighters are PERMITTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] (world_the_end) Lava fire is PERMITTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] (world_the_end) Fire spread is UNRESTRICTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] Loaded configuration for world 'world_the_end'
[19:02:16] [Server thread/INFO]: [WorldGuard] (Lobby) TNT ignition is PERMITTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] (Lobby) Lighters are PERMITTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] (Lobby) Lava fire is PERMITTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] (Lobby) Fire spread is UNRESTRICTED.
[19:02:16] [Server thread/INFO]: [WorldGuard] Loaded configuration for world 'Lobby'
[19:02:16] [Server thread/INFO]: [WorldGuard] Loading region data...
[19:02:16] [Server thread/INFO]: [Citizens] Enabling Citizens v2.0.28-SNAPSHOT (build 2346)
[19:02:16] [Server thread/ERROR]: [Citizens] Missing translation for system language (tr_TR): defaulting to English
[19:02:17] [Server thread/INFO]: [LockettePro] Enabling LockettePro v2.11.1
[19:02:17] [Server thread/INFO]: [LockettePro] Compatible server version detected: 1.17.1
[19:02:17] [Server thread/INFO]: [LockettePro] ===================================
[19:02:17] [Server thread/INFO]: [TownyCultures] Enabling TownyCultures v2.0.1
[19:02:17] [Server thread/INFO]:  .---.                           .--.     . . 
[19:02:17] [Server thread/INFO]:    |                            :         |_|_
[19:02:17] [Server thread/INFO]:    | .-..  .    ._.--. .  .     |    .  . | |  .  . .--..-. .--.
[19:02:17] [Server thread/INFO]:    |(   )\  \  /  |  | |  |     :    |  | | |  |  | |  (.-' `--.
[19:02:17] [Server thread/INFO]:    ' `-'  `' `'   '  `-`--|      `--'`--`-`-`-'`--`-'   `--'`--'
[19:02:17] [Server thread/INFO]:                           ;
[19:02:17] [Server thread/INFO]:                        `-'
[19:02:17] [Server thread/INFO]:                         by Goosius & LlmDl
[19:02:17] [Server thread/INFO]: 
[19:02:17] [Server thread/INFO]: [TownyCultures] Towny version 0.100.2.6 found.
[19:02:17] [Server thread/ERROR]: Error occurred while enabling TownyCultures v2.0.1 (Is it up to date?)
java.lang.NoClassDefFoundError: com/palmergames/bukkit/towny/exceptions/initialization/TownyInitException
	at com.gmail.goosius.townycultures.TownyCultures.onEnable(TownyCultures.java:53) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:561) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:475) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:733) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:317) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1220) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[patched_1.17.1.jar:git-Paper-411]
	at java.lang.Thread.run(Thread.java:842) ~[?:?]
Caused by: java.lang.ClassNotFoundException: com.palmergames.bukkit.towny.exceptions.initialization.TownyInitException
	at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:151) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:103) ~[patched_1.17.1.jar:git-Paper-411]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	... 11 more
[19:02:17] [Server thread/INFO]: [TownyCultures] Disabling TownyCultures v2.0.1
[19:02:17] [Server thread/ERROR]: [TownyCultures] Shutting down....
[19:02:17] [Server thread/INFO]: [BetterRTP] Enabling BetterRTP v3.6.12
[19:02:18] [Server thread/INFO]: [BetterRTP] Cooldown = 600
[19:02:18] [Server thread/INFO]: [QuickShop] Enabling QuickShop v5.1.2.5
[19:02:18] [Server thread/INFO]: [QuickShop] QuickShop Reremake
[19:02:18] [Server thread/INFO]: [QuickShop] Starting plugin self-test, please wait...
[19:02:18] [Server thread/INFO]: [QuickShop] [OK] Signature Verify
[19:02:18] [Server thread/INFO]: [QuickShop] [OK] Java Runtime Environment Version Test
[19:02:18] [Server thread/INFO]: [QuickShop] [OK] Potential Infection Characteristics Check
[19:02:18] [Server thread/INFO]: [QuickShop] [OK] Plugin Manifest Check
[19:02:18] [Server thread/INFO]: [QuickShop] Running QuickShop-Reremake on NMS version v1_17_R1 For Minecraft version 1.17.1
[19:02:18] [Server thread/INFO]: [QuickShop] [OK] Spigot Based Server Test
[19:02:18] [Server thread/INFO]: [QuickShop] [OK] Old QuickShop Test
[19:02:18] [Server thread/INFO]: [QuickShop] [OK] ModdedServer Based Test
[19:02:18] [Server thread/INFO]: [QuickShop] [OK] CoreSupport Test
[19:02:19] [Server thread/INFO]: [QuickShop] [OK] Virtual DisplayItem Support Test
[19:02:19] [Server thread/INFO]: [QuickShop] [OK] GameVersion supporting Test
[19:02:19] [Server thread/INFO]: [QuickShop] [OK] PacketListenerAPI Conflict Test
[19:02:19] [Server thread/INFO]: [QuickShop] [OK] End of life Test
[19:02:19] [Server thread/INFO]: [QuickShop] Reading the configuration...
[19:02:19] [Server thread/INFO]: [QuickShop] Developers: PotatoCraft Studio, Netherfoam, Timtower, KaiNoMood (KaiKikuchi), jho5245, Ghost_chu, cakoyo, Ectabro, portlek, log4b0at, Andre601, deadman96385, Vlvxingze, DoctaEnkoda, Mgazul, TiaraRinne, sandtechnology, Starmism, Chris6ix, Rean Schwarzer, mart-r, raphtaliapt, Tim269, creatorfromhell, LoneDev6, judgetread, confuxeon, ibmibmibm, yannicklamprecht, PyvesB, PaulBGD, ORelio, JoschuaSchneider, Starmium, harry0198, efekurbann, tdiant
[19:02:19] [Server thread/INFO]: [QuickShop] Original author: Netherfoam, Timtower, KaiNoMood
[19:02:19] [Server thread/INFO]: [QuickShop] Let's start loading the plugin
[19:02:19] [Server thread/INFO]: [QuickShop] Chat processor selected: Hardcoded BungeeChat Lib
[19:02:19] [Server thread/INFO]: [QuickShop] Loading plugin translations files...
[19:02:19] [Server thread/INFO]: [QuickShop] Game assets server selected: Mojang API
[19:02:19] [Server thread/INFO]: [QuickShop] Loading items translations...
[19:02:19] [Server thread/INFO]: [QuickShop] Loading enchantments translations...
[19:02:19] [Server thread/INFO]: [QuickShop] Loading potions translations...
[19:02:20] [Server thread/INFO]: [QuickShop] Successfully loaded WorldEdit support!
[19:02:20] [Server thread/WARN]: [QuickShop] You're using Real Display system and that may cause your server lagg, switch to Virtual Display system if you can! (Checkout display-type option in config.yml)
[19:02:20] [Server thread/INFO]: [QuickShop] Setting up database...
[19:02:20] [Server thread/WARN]: [QuickShop] Loaded class fr.xephi.authme.libs.com.mysql.cj.jdbc.Driver from AuthMe v5.6.0-SNAPSHOT-b2623 which is not a depend, softdepend or loadbefore of this plugin.
[19:02:20] [Server thread/INFO]: [QuickShop] Checking and updating database columns, it may take a while...
[19:02:20] [Server thread/INFO]: [QuickShop] Finished!
[19:02:20] [Server thread/INFO]: [QuickShop] Selected permission provider: Bukkit
[19:02:20] [Server thread/INFO]: [QuickShop] Registering commands...
[19:02:20] [Server thread/INFO]: [QuickShop] Loaded 1 rules for listener blacklist.
[19:02:20] [Server thread/INFO]: [QuickShop] EventManager selected: QSEventManager
[19:02:20] [Server thread/INFO]: [QuickShop] Fetching shops from the database...If plugin stuck there, check your database connection.
[19:02:20] [Server thread/INFO]: [QuickShop] Loading shops from the database...
[19:02:20] [Server thread/INFO]: [QuickShop] >> Shop Loader Information
[19:02:20] [Server thread/INFO]: [QuickShop] Total           shops: 0
[19:02:20] [Server thread/INFO]: [QuickShop] Valid           shops: 0
[19:02:20] [Server thread/INFO]: [QuickShop] Pending              : 0
[19:02:20] [Server thread/INFO]: [QuickShop] Waiting worlds loaded: 0
[19:02:20] [Server thread/INFO]: [QuickShop] Waiting chunks loaded: 0
[19:02:20] [Server thread/INFO]: [QuickShop] Done! Used 3ms to loaded shops in database.
[19:02:20] [Server thread/INFO]: [QuickShop] Registering listeners...
[19:02:20] [Server thread/INFO]: [QuickShop] Registering DisplayCheck task....
[19:02:20] [Server thread/INFO]: [QuickShop] Cleaning MsgUtils...
[19:02:20] [Server thread/INFO]: [QuickShop] Cleaning purchase messages from the database that are over a week old...
[19:02:20] [Server thread/INFO]: [QuickShop] Log actions is enabled, actions will log in the qs.log file!
[19:02:20] [Server thread/INFO]: [QuickShop] [Shop Purger] Purge not enabled!
[19:02:20] [Server thread/INFO]: [QuickShop] QuickShop Loaded! 2156 ms.
[19:02:20] [Server thread/INFO]: [SiegeWar] Enabling SiegeWar v2.10.0
[19:02:20] [Server thread/ERROR]: Error occurred while enabling SiegeWar v2.10.0 (Is it up to date?)
java.lang.NoClassDefFoundError: com/palmergames/bukkit/util/Colors
	at com.gmail.goosius.siegewar.SiegeWar.printSickASCIIArt(SiegeWar.java:210) ~[?:?]
	at com.gmail.goosius.siegewar.SiegeWar.onEnable(SiegeWar.java:72) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:561) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:475) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:733) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:317) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1220) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[patched_1.17.1.jar:git-Paper-411]
	at java.lang.Thread.run(Thread.java:842) ~[?:?]
Caused by: java.lang.ClassNotFoundException: com.palmergames.bukkit.util.Colors
	at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:151) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:103) ~[patched_1.17.1.jar:git-Paper-411]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	... 12 more
[19:02:20] [Server thread/INFO]: [SiegeWar] Disabling SiegeWar v2.10.0
[19:02:20] [Server thread/INFO]: [SiegeWar] Shutting down...
[19:02:20] [Server thread/INFO]: [TownyFlight] Enabling TownyFlight v1.10.2
[19:02:20] [Server thread/ERROR]: Error occurred while enabling TownyFlight v1.10.2 (Is it up to date?)
java.lang.NoClassDefFoundError: com/palmergames/bukkit/config/CommentedConfiguration
	at com.gmail.llmdlio.townyflight.config.TownyFlightConfig.loadConfig(TownyFlightConfig.java:35) ~[?:?]
	at com.gmail.llmdlio.townyflight.config.TownyFlightConfig.reload(TownyFlightConfig.java:20) ~[?:?]
	at com.gmail.llmdlio.townyflight.TownyFlight.loadConfig(TownyFlight.java:95) ~[?:?]
	at com.gmail.llmdlio.townyflight.TownyFlight.loadSettings(TownyFlight.java:89) ~[?:?]
	at com.gmail.llmdlio.townyflight.TownyFlight.onEnable(TownyFlight.java:49) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:561) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:475) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:733) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:317) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1220) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[patched_1.17.1.jar:git-Paper-411]
	at java.lang.Thread.run(Thread.java:842) ~[?:?]
Caused by: java.lang.ClassNotFoundException: com.palmergames.bukkit.config.CommentedConfiguration
	at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:151) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:103) ~[patched_1.17.1.jar:git-Paper-411]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	... 15 more
[19:02:20] [Server thread/INFO]: [TownyFlight] Disabling TownyFlight v1.10.2
[19:02:20] [Server thread/INFO]: Running delayed init tasks
[19:02:20] [Server thread/INFO]: [QuickShop] Registering bStats metrics...
[19:02:20] [Server thread/INFO]: [GroupManager] Bukkit Permissions Updated!
[19:02:21] [Craft Scheduler Thread - 5 - AuthMe/INFO]: [AuthMe] Downloading GEO IP database, because the old database is older than 30 days or doesn't exist
[19:02:21] [Craft Scheduler Thread - 5 - AuthMe/WARN]: [AuthMe] No MaxMind credentials found in the configuration file! GeoIp protections will be disabled.
[19:02:21] [Craft Scheduler Thread - 5 - AuthMe/INFO]: [AuthMe] There is no newer GEO IP database uploaded to MaxMind. Using the old one for now.
[19:02:21] [Craft Scheduler Thread - 5 - AuthMe/WARN]: [AuthMe] Could not download GeoLiteAPI database [FileNotFoundException]: plugins\AuthMe\GeoLite2-Country.mmdb (Sistem belirtilen dosyayı bulamıyor)
[19:02:21] [Craft Scheduler Thread - 13 - Vault/INFO]: [Vault] Checking for Updates ... 
[19:02:21] [Server thread/INFO]: [Citizens] Loaded 2 NPCs.
[19:02:21] [Server thread/ERROR]: [QuickShop] No Economy plugin detected! Please make sure that you have a
[19:02:21] [Server thread/ERROR]: [QuickShop] compatible economy plugin installed to get Vault working.
[19:02:21] [Server thread/INFO]: Done (50.349s)! For help, type "help"
[19:02:21] [Server thread/INFO]: Timings Reset
[19:02:21] [Craft Scheduler Thread - 10 - GroupManager/INFO]: [GroupManager] No new version available
[19:02:21] [Craft Scheduler Thread - 13 - Vault/INFO]: [Vault] No new version available
[19:02:21] [Craft Scheduler Thread - 4 - LagAssist/INFO]: LagAssist » You are up to date with LagAssist.
```
3. Your Towny config.yml: I didn't change anything.
4. Your townyperms.yml:Not loaded
5. Your log's error:
```
[19:02:15] [Server thread/INFO]: ====================      Towny      ========================
[19:02:15] [Server thread/ERROR]: Error occurred while enabling Towny v0.100.2.6 (Is it up to date?)
java.lang.NoClassDefFoundError: org/apache/commons/io/FilenameUtils
	at com.palmergames.compress.utils.FileNameUtils.getBaseName(FileNameUtils.java:48) ~[?:?]
	at com.palmergames.bukkit.towny.object.TranslationLoader.lambda$getLangFileNamesFromPlugin$0(TranslationLoader.java:196) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178) ~[?:?]
	at java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?]
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?]
	at com.palmergames.bukkit.towny.object.TranslationLoader.getLangFileNamesFromPlugin(TranslationLoader.java:197) ~[?:?]
	at com.palmergames.bukkit.towny.object.TranslationLoader.loadTranslationsIntoMemory(TranslationLoader.java:164) ~[?:?]
	at com.palmergames.bukkit.towny.object.Translation.loadTranslationRegistry(Translation.java:47) ~[?:?]
	at com.palmergames.bukkit.towny.Towny.loadLocalization(Towny.java:276) ~[?:?]
	at com.palmergames.bukkit.towny.Towny.loadFoundation(Towny.java:227) ~[?:?]
	at com.palmergames.bukkit.towny.Towny.onEnable(Towny.java:137) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:561) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:475) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:733) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:317) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1220) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[patched_1.17.1.jar:git-Paper-411]
	at java.lang.Thread.run(Thread.java:842) ~[?:?]
Caused by: java.lang.ClassNotFoundException: org.apache.commons.io.FilenameUtils
	at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:151) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:103) ~[patched_1.17.1.jar:git-Paper-411]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	... 27 more
[19:02:15] [Server thread/INFO]: [Towny] Disabling Towny v0.100.2.6
[19:02:15] [Server thread/INFO]: ==============================================================
[19:02:15] [Server thread/INFO]: [Towny] Finishing Universe Tasks...
[19:02:15] [Server thread/INFO]: [Towny] Version: 0.100.2.6 - Plugin Disabled
```

> > It didn't work but thanks.
> 
> Do you have an error in your log you can share

This is the error in the file you gave.

[20:55:18] [Server thread/INFO]: [Towny] Enabling Towny v0.100.2.7
[20:55:18] [Server thread/INFO]: ====================      Towny      ========================
[20:55:19] [Server thread/ERROR]: Error occurred while enabling Towny v0.100.2.7 (Is it up to date?)
java.lang.NoClassDefFoundError: org/apache/commons/io/FilenameUtils
	at com.palmergames.compress.utils.FileNameUtils.getBaseName(FileNameUtils.java:66) ~[?:?]
	at com.palmergames.util.FileMgmt.getBaseFileName(FileMgmt.java:493) ~[?:?]
	at com.palmergames.bukkit.towny.object.TranslationLoader.lambda$getLangFileNamesFromPlugin$0(TranslationLoader.java:196) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178) ~[?:?]
	at java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?]
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?]
	at com.palmergames.bukkit.towny.object.TranslationLoader.getLangFileNamesFromPlugin(TranslationLoader.java:197) ~[?:?]
	at com.palmergames.bukkit.towny.object.TranslationLoader.loadTranslationsIntoMemory(TranslationLoader.java:164) ~[?:?]
	at com.palmergames.bukkit.towny.object.Translation.loadTranslationRegistry(Translation.java:47) ~[?:?]
	at com.palmergames.bukkit.towny.Towny.loadLocalization(Towny.java:276) ~[?:?]
	at com.palmergames.bukkit.towny.Towny.loadFoundation(Towny.java:227) ~[?:?]
	at com.palmergames.bukkit.towny.Towny.onEnable(Towny.java:137) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:561) ~[patched_1.17.1.jar:git-Paper-411]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:475) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:733) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:317) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1220) ~[patched_1.17.1.jar:git-Paper-411]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[patched_1.17.1.jar:git-Paper-411]
	at java.lang.Thread.run(Thread.java:842) ~[?:?]
Caused by: java.lang.ClassNotFoundException: org.apache.commons.io.FilenameUtils
	... 28 more
[20:55:19] [Server thread/INFO]: [Towny] Disabling Towny v0.100.2.7
[20:55:19] [Server thread/INFO]: ==============================================================
[20:55:19] [Server thread/INFO]: [Towny] Finishing Universe Tasks...
[20:55:19] [Server thread/INFO]: [Towny] Version: 0.100.2.7 - Plugin Disabled
[20:55:19] [Server thread/INFO]: =============================================================

--- a/Towny/src/main/java/com/palmergames/bukkit/towny/object/TranslationLoader.java
+++ b/Towny/src/main/java/com/palmergames/bukkit/towny/object/TranslationLoader.java
@@ -25,7 +25,6 @@ import java.util.logging.Level;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import org.apache.commons.compress.utils.FileNameUtils;
 import org.bukkit.plugin.Plugin;
 import org.yaml.snakeyaml.LoaderOptions;
 import org.yaml.snakeyaml.Yaml;
@@ -193,8 +192,9 @@ public class TranslationLoader {
 			uri = clazz.getResource("").toURI();
 			
 			try (final FileSystem fs = FileSystems.newFileSystem(uri, Collections.emptyMap()); Stream<Path> stream  = Files.list(fs.getRootDirectories().iterator().next().resolve("/lang"))) {
-				stream.filter(p -> TownySettings.isLanguageEnabled(FileNameUtils.getBaseName(p)))
-					.forEach(p -> lang.add(FileNameUtils.getBaseName(p)));
+				stream.map(FileMgmt::getFileName)
+					.filter(TownySettings::isLanguageEnabled)
+					.forEach(lang::add);
 			}
 		} catch (URISyntaxException | IOException e) {
 			plugin.getLogger().log(Level.WARNING, "An exception occurred while getting language file names from the plugin jar", e);
@@ -247,11 +247,11 @@ public class TranslationLoader {
 		File[] overrideFiles = new File(langFolderPath + File.separator + "override").listFiles();
 		if (overrideFiles != null) {
 			for (File file : overrideFiles) {
-				if (file.isFile() && FileNameUtils.getExtension(file.toPath()).equalsIgnoreCase("yml") 
-					&& !file.getName().equalsIgnoreCase("global.yml") && TownySettings.isLanguageEnabled(FileNameUtils.getBaseName(file.toPath()))) {
+				if (file.isFile() && FileMgmt.getExtension(file.toPath()).equalsIgnoreCase("yml") 
+					&& !file.getName().equalsIgnoreCase("global.yml") && TownySettings.isLanguageEnabled(FileMgmt.getFileName(file.toPath()))) {
 					try (FileInputStream is = new FileInputStream(file)) {
 						Map<String, Object> values = new Yaml(new SafeConstructor(new LoaderOptions())).load(is);
-						String lang = FileNameUtils.getBaseName(file.toPath()).replaceAll("-", "_");
+						String lang = FileMgmt.getFileName(file.toPath()).replaceAll("-", "_");
 
 						if (values != null) {
 							newTranslations.computeIfAbsent(lang, k -> new HashMap<>());

--- a/Towny/src/main/java/com/palmergames/util/FileMgmt.java
+++ b/Towny/src/main/java/com/palmergames/util/FileMgmt.java
@@ -7,6 +7,8 @@ import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
 import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
 import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream;
 import org.apache.commons.io.IOUtils;
+import org.jetbrains.annotations.NotNull;
+
 import java.io.BufferedWriter;
 import java.io.DataOutputStream;
 import java.io.File;
@@ -477,5 +479,29 @@ public final class FileMgmt {
 		try (FileOutputStream fos = new FileOutputStream(path.toFile())) {
 			fos.write(string.getBytes(StandardCharsets.UTF_8));
 		}
+	}
+
+	/**
+	 * @param path The file path
+	 * @return The file name for the given path (minus file extension)
+	 */
+	@NotNull
+	public static String getFileName(final @NotNull Path path) {
+		String fileName = path.getFileName().toString();
+		if (fileName.contains("."))
+			fileName = fileName.substring(0, fileName.lastIndexOf("."));
+
+		return fileName;
+	}
+	
+	@NotNull
+	public static String getExtension(final @NotNull Path path) {
+		String fileName = path.getFileName().toString();
+		
+		final int index = fileName.lastIndexOf(".");
+		if (index == -1)
+			return "";
+		
+		return fileName.substring(index + 1);
 	}
 }

